### PR TITLE
Feature/improve matrix 2d

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/components/general/OrientationInput/styles.scss [hunk 1]
```diff
@@ -1,4 +1,6 @@
 .orientation-input {
+    min-width: 64px;
```
Comment (AdityaKhatri) [inline src/components/general/OrientationInput/styles.scss:2]: Plz move this

File: src/widgetComponents/Matrix2dInput/index.js [hunk 2]
```diff
@@ -193,59 +207,80 @@ export default class Matrix2dInput extends React.PureComponent {
...
                                     rendererParams={this.subsectorTitleRendererParams}
                                 />
                             </tr>
                         </thead>
                     ) : (
                         <thead>
+                            <tr>
+                                <th
+                                    className={styles.topTh}
+                                    colSpan={sectors.length + 2}
```
Comment (AdityaKhatri) [inline src/widgetComponents/Matrix2dInput/index.js]: Why though? Plz leave note

File: src/widgets/edit/Matrix2d/index.js [hunk 3]
```diff
@@ -484,41 +496,105 @@ export default class Matrix2dEditWidget extends React.PureComponent {
...
+                                        label={_ts('widgets.editor.matrix2d', 'titleRowHeightLabel')}
+                                        type="number"
+                                        placeholder={_ts('widgets.editor.matrix2d', 'heightInputPlaceholder')}
+                                        persistantHintAndError={false}
+                                    />
+                                    <TextInput
+                                        className={styles.fontSizeInput}
+                                        faramElementName="titleRowFontSize"
+                                        label={_ts('widgets.editor.matrix2d', 'titleRowFontSizeLabel')}
+                                        type="number"
```
Comment (AdityaKhatri) [inline src/widgets/edit/Matrix2d/index.js]: Use number input

File: src/components/general/Navbar/NavMenu/index.js [hunk 4]
```diff
@@ -54,7 +59,7 @@ const defaultProps = {
 };
 
 @withRouter
-export default class NavMenu extends React.PureComponent {
+class NavMenu extends React.PureComponent {
```
Comment (tnagorra) [inline src/components/general/Navbar/NavMenu/index.js:61]: We should remove withRouter here

File: src/views/EditEntries/List/index.js [hunk 5]
```diff
@@ -59,13 +59,33 @@ export default class Listing extends React.PureComponent {
         window.addEventListener('scroll', this.handleScroll, true);
     }
 
+    componentDidUpdate(prevProps) {
+        if (this.scrollTop && prevProps.hash !== this.props.hash && this.props.hash === '#/list') {
```
Comment (tnagorra) [inline src/views/EditEntries/List/index.js]: `this.scrollTop` should be `isTruthy(this.scrollTop)`

**Pull request merged**